### ISSUE TITLE
Using fixmatch to train on my dataset, LR = 0.03, batchsize = 64. When the unlabeled loss increases gradually, the model gets worse and worse, and the loss of training set will decrease, but the loss of verification set will increase, and the accuracy will decrease?

### ISSUE THREAD

**Issue by a571-op** [kekmodel/FixMatch-pytorch#23]
Using fixmatch to train on my dataset, LR = 0.03, batchsize = 64. When the unlabeled loss increases gradually, the model gets worse and worse, and the loss of training set will decrease, but the loss of verification set will increase, and the accuracy will decrease?

**Comment by KittyStore** [comment]
In my opinion, the reason that had caused this problem is as follows : In the training phase, the test function is called, and the test function makes the model enter the eval mode. When the training continues, it does not switch to the train mode, so only the first batch of data uses BN and Dropout, which leads to the decline of accuracy. Maybe putting model.train () into every epoch loop will solve this question.Hope to help you.

**Comment by cjc20000323** [comment]
Have you understood the reason?
